gate-web/src/main/groovy/com/netflix/spinnaker/gate/controllers/PipelineController.groovy

```diff
@@ -119,10 +119,14 @@ class PipelineController {
     ]
 
     def result = taskService.createAndWaitForCompletion(operation)
+    String resultStatus = result.get("status")
 
-    if ("TERMINAL".equalsIgnoreCase((String) result.get("status"))) {
+    if ("TERMINAL".equalsIgnoreCase(resultStatus)) {
```

Comment:
I am inclined to keep the terminal check for a while to see if there's some other non-success status coming back.